doc/chapters/filesystem/index.rst

```diff
@@ -0,0 +1,102 @@
+Filesystems
+***********
+
+This chapter documents the behavior and layout of the basic filesystem of the
+containers and where it picks up the different packages from. 
+
+
+Standard filesystem
+===================
+
+The base filesystem is bind mounted from a directory created by lxc, the
+directory is empty by default and is setup by the ``lxc-softwarecontainer.in``
+script, which is run by lxc init. 
+
+* The script creates several basic directories creating a normal unix
+  environment as defined by Filesystem Hierarchy Standard (FHS). 
+* The filesystem is still empty except for directories. 
+* LXC then mounts some basic directories defined in the
+  ``libsoftwarecontainer/softwarecontainer.conf`` (copied to
+  ``/usr/share/lxc/conf`` on installation). 
+* On top of this, non standard mountpoints are also used to mount for example
+  the gateway directories etc inside the filesystem. These are defined using
+  BindMountFolderInContainer, or mounted by the different gateways.
+
+The basic filesystem looks like this on startup::
+
+	.
+	├── config
+	└── rootfs
+	    ├── bin
+	    ├── dev
+	    ├── etc
+	    │   ├── group
+	    │   ├── machine-id
+	    │   ├── passwd
+	    │   ├── pulse
+	    │   │   └── client.conf
+	    │   └── resolv.conf
+	    ├── gateways
+	    ├── home
+	    ├── lib
+	    ├── lib64
+	    ├── proc
+	    ├── root
+	    ├── sbin
+	    ├── tmp
+	    └── usr
+	       	├── lib
+	       	├── lib64
+	       	├── local
+	       	└── sbin
+
+
+Mountpoints
+===========
+
+On top of the above structure, mount points can be created anywhere inside the
+container filestructure using the BindMountFolderInContainer and gateway
```

Comment:
It might be good to mention here that BindmountFolderInContainer is an API call.